akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala

```diff
@@ -40,56 +40,78 @@ trait ActorContext[T] {
 
   /**
    * Get the `scaladsl` of this `ActorContext`.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as CompletionStage callbacks.
    */
   def asScala: akka.actor.typed.scaladsl.ActorContext[T]
 
   /**
    * The identity of this Actor, bound to the lifecycle of this Actor instance.
    * An Actor with the same name that lives before or after this instance will
    * have a different [[ActorRef]].
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as CompletionStage callbacks.
    */
   def getSelf: ActorRef[T]
 
-  /**
-   * Return the mailbox capacity that was configured by the parent for this actor.
-   */
-  def getMailboxCapacity: Int
-
   /**
    * The [[ActorSystem]] to which this Actor belongs.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as CompletionStage callbacks.
    */
   def getSystem: ActorSystem[Void]
 
   /**
    * The list of child Actors created by this Actor during its lifetime that
    * are still alive, in no particular order.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
+   * actor message processing thread, such as CompletionStage callbacks.
    */
   def getChildren: java.util.List[ActorRef[Void]]
 
   /**
    * The named child Actor if it is alive.
+   *
+   * This method is thread-safe and can be called from other threads than the ordinary
```

Comment:
is it really thread safe? perhaps, but I'd not recommend touching anything related to children from other threads. It could at least be confusing stale information.

Reply:
Yeah, I _think_ they are safe, volatile field and snapshots/defensive copies as far as I can see, but I agree, that's not the same as getting what you expect back or recommending using them from other threads.

Reply:
I thought about this some, and decided that even if it technically is thread safe, we might not want to guarantee that, and I didn't want to write a somewhat long winded "this is technically safe but is internal state to the actor so don't do that" so instead I marked the children accessor as non thread safe as well to nudge users in the right direction.